src/OpenTelemetry.Instrumentation.Runtime/RuntimeMetrics.cs

```diff
@@ -69,31 +69,33 @@ public RuntimeMetrics(RuntimeMetricsOptions options)
 #if NET6_0_OR_GREATER
             if (options.IsJitEnabled)
             {
-                this.meter.CreateObservableCounter($"{metricPrefix}il.bytes.jitted", () => System.Runtime.JitInfo.GetCompiledILBytes(), "By", description: "IL Bytes Jitted.");
-                this.meter.CreateObservableCounter($"{metricPrefix}methods.jitted.count", () => System.Runtime.JitInfo.GetCompiledMethodCount(), description: "Number of Methods Jitted.");
-                this.meter.CreateObservableCounter($"{metricPrefix}time.in.jit", () => System.Runtime.JitInfo.GetCompilationTime().Ticks * NanosecondsPerTick, "ns", description: "Time spent in JIT.");
+                this.meter.CreateObservableCounter($"{metricPrefix}jit.il_compiled", () => System.Runtime.JitInfo.GetCompiledILBytes(), unit: "By", description: "Count of bytes of intermediate language that have been compiled since the process start.");
+                this.meter.CreateObservableCounter($"{metricPrefix}jit.methods_compiled", () => System.Runtime.JitInfo.GetCompiledMethodCount(), description: "Count of methods that have been compiled since the process start.");
+                this.meter.CreateObservableCounter($"{metricPrefix}jit.compilation_time", () => System.Runtime.JitInfo.GetCompilationTime().Ticks * NanosecondsPerTick, unit: "ns", description: "The amount of time the JIT compiler has spent compiling methods since the process start.");
```

Comment:
OpenTelemetry in general uses `s` as the unit, for example https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/system-metrics.md#metric-instruments.

Here I think `s` (double) might cause problems such as drifting, precision loss and saturation https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/supplementary-guidelines.md#float.

For microsec or millisec, do you think an integer would be good, or we need to have float64? If we're okay with integer (and the precision loss for the fraction of micro/milli sec), I think it's doable.

Regarding `ns` as the unit (for int64), I think the #1 concern is int overflow, and it seems to be okay since int64.max_value nanoseconds equal to `9223372036.85` seconds, which is large enough (even for a small data center of servers)?

Reply:
I think there are other places where we see struggle with units. We can follow up in another PR.